dspace-api/src/main/java/org/dspace/qaevent/service/BrokerClientFactory.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.qaevent.service;
+
+import eu.dnetlib.broker.BrokerClient;
+import org.dspace.utils.DSpace;
+
+/**
+ * Factory for the {@link BrokerClient}.
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ *
+ */
+public interface BrokerClientFactory {
```

Comment:
Yes it is openaire specific but consider we are going to add similar java classes for all the sources to integrate. At time of writing, for _this_ pr, the source involved is just **openaire**. Looking at this pr evolution at [https://github.com/DSpace/DSpace/pull/9218](https://github.com/DSpace/DSpace/pull/9218) you see we involve Coar as a new source: figure it as a new source right next to Openaire.

Reply:
@frabacche : Maybe I'm misunderstanding, but I still only see _one implementation_ of `BrokerClientFactory` interface in #9218 and it looks to be OpenAIRE specific.

I do understand that we'll want similar classes for other sources like COAR Notify.  But, my point is that it appears (to me) that `BrokerClientFactory` itself can only currently work for OpenAIRE as it uses the OpenAIRE API in `BrokerClientFactoryImpl`.  If I'm wrong and there is a COAR implementation of BrokerClientFactory, then please let me know.  Otherwise, I'd recommend naming this `OpenaireBrokerClientFactory` if it's only ever used by OpenAIRE.